go/types/leaf_sequence.go

```diff
@@ -72,10 +72,11 @@ func (seq leafSequence) valuesSlice(from, to uint64) []Value {
 	if len := seq.Len(); to > len {
 		to = len
 	}
+
 	dec := seq.decoderSkipToIndex(int(from))
-	vs := make([]Value, to-from)
-	for i := from; i < to; i++ {
-		vs[i-from] = dec.readValue()
+	vs := make([]Value, (to-from)*getValuesPerIdx(seq))
```

Comment:
Updating this made me realise that `values()` had a bug for Maps, we just never called it for Maps.